test/models/meshgraphnet/test_meshgraphnet_snmg.py

```diff
@@ -125,13 +129,18 @@ def run_test_distributed_meshgraphnet(rank, world_size, dtype):
     loss.backward()
 
     out_multi_gpu = model_multi_gpu(nfeat_multi_gpu, efeat_multi_gpu, graph_multi_gpu)
-    loss = out_multi_gpu.sum()
+    # PyTorch unfortunately averages across all process groups within DistributedDataParallel
+    # by default, for tensor-parallel applications like this one, potential solutions
+    # are either custom gradient hooks (the best solution for actual training workloads)
+    # or multiplying by world_size to cancel out the normalization. As this is just a simple
+    # test, we do the easier thing here as we only compare the weight gradients.
+    loss = out_multi_gpu.sum() * world_size
```

Comment:
Is this not an artifact of the fact that you're using DDP with `process_group=manager.group("graph_partition")` and also creating the distributed graph using `partition_group_name="graph_partition"`? Ideally, wouldn't it be that you set up DDP with `process_group=manager.group("orthogonal_graph_partition")`?

Reply:
IIUC, no. I would use the orthogonal process group if the the weights in each process group in `graph_partition` are not shared, right? Then you just would accumulate gradients for the same partition rank across different data-parallel groups.
Here, the model is shared and in a rather tensor-parallel setting. So the main artifact should stem from the fact that DDP by default averages the gradients based on world_size. My solution in actual code would be a custom gradient hook - though I wanted to keep the tests simpler and not deal with that.

Reply:
When the weights are shared, I was envisioning that you would instead do the all-reduce within the process group as a no-op autograd function that performs an all-reduce in the backward pass. But what you're saying is a valid approach too. This is another aspect to model parallel training that to me seems challenging to generalize. Would be good to think about this in more detail after getting this PR merged in.